### PR TITLE
Sanitize argument forwarding between "follow_link" and "links"

### PR DESCRIPTION
**Pull request by amotl**
Hi Dan,

this is a quick attempt to address #177 we just talked about.

With kind regards,
Andreas.

### CODE REVIEW

File: mechanicalsoup/stateful_browser.py [hunk 1]
```diff
@@ -204,17 +204,29 @@ def list_links(self, *args, **kwargs):
         for l in self.links(*args, **kwargs):
             print("    ", l)
 
-    def links(self, url_regex=None, link_text=None, *args, **kwargs):
+    def links(self, *args, **kwargs):
```
Comment (moy) [inline mechanicalsoup/stateful_browser.py]: Unfortunately this breaks backward compatibility for people using `url_regex` as a positional argument.

**Comment by amotl** [comment]
Dear Matthieu,

thanks for your response. I will try to find some time to address your request soon.

With kind regards,
Andreas.

**Comment by moy** [comment]
Hi,

Happy new year and a ping on this PR: @amotl: did you get time to address the issues in this PR?

Thanks,

**Comment by amotl** [comment]
Dear Matthieu,

also happy new year to you and your loved ones!

I am sorry to say that i haven't been able to address your suggestions yet. As quite some things are on my plate already, i can't promise to get to it in January, but i will be happy to receive another ping from you in February.

Otherwise: If others from the community read this and have some spare time, i will also be happy to pass this along.

With kind regards,
Andreas.

**Comment by moy** [comment]
This PR was doing too many things in one (for example the header-related changes are independent, and probably no longer needed as we now natively set the Referer, and it turns out other parts of the patch are not needed to fix the issue). I picked the important part, added tests, and included it in #184, so I'm closing this one.

**Comment by amotl** [comment]
All right @moy, thanks for taking over!

**Pull request closed without merging**